src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/generalletter/GeneralLetterService.java

```diff
@@ -0,0 +1,160 @@
+
+    private final CaseDetailsConverter caseDetailsConverter;
+    private final DocAssemblyService docAssemblyService;
+    private final ApplicationEventPublisher publisher;
+    private final DocumentManagementService documentManagementService;
+    private final Clock clock;
+
+    public GeneralLetterService(
+        CaseDetailsConverter caseDetailsConverter,
+        DocAssemblyService docAssemblyService,
+        ApplicationEventPublisher publisher,
+        DocumentManagementService documentManagementService,
+        Clock clock
+    ) {
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.docAssemblyService = docAssemblyService;
+        this.publisher = publisher;
+        this.documentManagementService = documentManagementService;
+        this.clock = clock;
+    }
+
+    public CallbackResponse createAndPreview(CaseDetails caseDetails, String authorisation,
+                                             String letterType, String templateId) {
+        try {
+            logger.info("General Letter: creating letter");
+            CCDCase ccdCase = caseDetailsConverter.extractCCDCase(caseDetails);
+            DocAssemblyResponse docAssemblyResponse = docAssemblyService.createGeneralLetter(ccdCase,
+                authorisation, templateId);
+            return AboutToStartOrSubmitCallbackResponse
+                .builder()
+                .data(ImmutableMap.of(
+                    letterType,
+                    CCDDocument.builder().documentUrl(docAssemblyResponse.getRenditionOutputLocation())))
+                .build();
+        } catch (Exception e) {
+            logger.info("General Letter creating and preview failed", e);
+            return AboutToStartOrSubmitCallbackResponse
+                .builder()
+                .errors(Collections.singletonList(ERROR_MESSAGE))
+                .build();
+        }
+    }
+
+    public CallbackResponse printAndUpdateCaseDocuments(CaseDetails caseDetails, String authorisation) {
+        try {
+            CCDCase ccdCase = caseDetailsConverter.extractCCDCase(caseDetails);
+            CCDDocument draftLetterDoc = ccdCase.getDraftLetterDoc();
+            Claim claim = caseDetailsConverter.extractClaim(caseDetails);
+            printLetter(authorisation, draftLetterDoc, claim);
+            CCDCase updatedCase = ccdCase.toBuilder()
+                .caseDocuments(updateCaseDocumentsWithOrder(ccdCase, draftLetterDoc))
+                .build();
+            return AboutToStartOrSubmitCallbackResponse
+                .builder()
+                .data(caseDetailsConverter.convertToMap(updatedCase))
+                .build();
+
+        } catch (Exception e) {
+            logger.info("General Letter printing and case documents update failed", e);
+            return AboutToStartOrSubmitCallbackResponse
+                .builder()
+                .errors(Collections.singletonList(ERROR_MESSAGE))
```

Comment:
We're not sure nothing has been sent in this case - the exception could have been thrown afterwards.

Reply:
discussed this with Chris and now only catching exception when fails on docmosis call or bulkprint.